### PR TITLE
Update Archive Project template

### PR DESCRIPTION
**Pull request by mrbobbytables**
Small amount of cleanup I saw when creating the archive issue for curiefense:
- remove amye, add mrbobbytables, castrojo
- remove unnecessary whitepad & duplicate task

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/archived.md [hunk 1]
```diff
@@ -3,19 +3,18 @@ name: Archiving checklist
...
 - [ ] Remove/Archive on maintainers.cncf.io
-- [ ] Remove logo from all slide decks	
-- [ ] Remove logo from event sites	
-- [ ] Move logo to archived in artwork	
-- [ ] Archive DevStats (or archive?)	
-- [ ] Notify Speakers team to remove from CFPs	
-- [ ] Remove from service desk or archive	
+- [ ] Remove logo from all slide decks
+- [ ] Remove logo from event sites
+- [ ] Archive DevStats
```
Comment (krook) [inline .github/ISSUE_TEMPLATE/archived.md:16]: Should add Insights here too, and potential other LFX tools

Comment (mrbobbytables) [reply .github/ISSUE_TEMPLATE/archived.md:16]: yeah, I noticed another thing. I'll flag this as WIP for now

**Comment by mrbobbytables** [comment]
Should be good for re-review :+1:

File: .github/ISSUE_TEMPLATE/archived.md [hunk 2]
```diff
@@ -3,19 +3,19 @@ name: Archiving checklist
...
+- [ ] Move project (logo) to archived in the [artwork repo](https://github.com/cncf/artwork)
+- [ ] Move project to archived on [landscape](https://github.com/cncf/landscape)
+- [ ] Remove maintainer list from [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv)
+- [ ] Archive (lock) maintainers mailing list (groups.io)
+- [ ] Remove ML from maintainers@ mailing group (google groups)
+- [ ] Remove project (logo) from all slide decks (contact pr@cncf.io)
+- [ ] Remove project (logo) from event sites (contact events@cncf.io)
+- [ ] Notify DevEx team to remove from event Maintainer Project Opportunities
+- [ ] Remove from maintainers & project from service desk
+- [ ] Remove from CNCF store (reach out to staff or emaill store@cncf.io)
```
Comment (krook) [inline .github/ISSUE_TEMPLATE/archived.md:18]: Small typo there... two Ls in email

Comment (mrbobbytables) [reply .github/ISSUE_TEMPLATE/archived.md]: thanks! fixed

File: .github/ISSUE_TEMPLATE/archived.md [hunk 3]
```diff
@@ -3,19 +3,19 @@ name: Archiving checklist
...
-- [ ] Move logo to archived in artwork	
-- [ ] Archive DevStats (or archive?)	
-- [ ] Notify Speakers team to remove from CFPs	
-- [ ] Remove from service desk or archive	
-- [ ] Remove from CNCF store
+- [ ] Move project (logo) to archived in the [artwork repo](https://github.com/cncf/artwork)
+- [ ] Move project to archived on [landscape](https://github.com/cncf/landscape)
+- [ ] Remove maintainer list from [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv)
+- [ ] Archive (lock) maintainers mailing list (groups.io)
+- [ ] Remove ML from maintainers@ mailing group (google groups)
```
Comment (kevin-wangzefeng) [inline .github/ISSUE_TEMPLATE/archived.md:13]: Nit: what is ML short for?

Comment (mrbobbytables) [reply .github/ISSUE_TEMPLATE/archived.md:13]: mailing list, the maintainers@ list does not have individuals but the maintainer mailing lists of every project:👍

**Review by jeefy — APPROVED** [review]

**Pull request merged**